room/compiler-processing/src/main/java/androidx/room/compiler/processing/javac/JavacRoundEnv.kt

```diff
@@ -47,4 +52,51 @@ internal class JavacRoundEnv(
             env.wrapTypeElement(MoreElements.asType(it))
         }.toSet()
     }
+
+    override fun getTypeElementsAnnotatedWith(klass: KClass<out Annotation>): Set<XTypeElement> {
+        return getTypeElementsAnnotatedWith(
+            annotationQualifiedName = klass.java.name
```

Comment:
i think we want `klass.java.canonicalName` here?
this seems to be the same for most cases but in some cases, it seems to return the binary representation?
From `.name` docs:

```
* <p> Examples:
     * <blockquote><pre>
     * String.class.getName()
     *     returns "java.lang.String"
     * byte.class.getName()
     *     returns "byte"
     * (new Object[3]).getClass().getName()
     *     returns "[Ljava.lang.Object;"
     * (new int[3][4][5][6][7][8][9]).getClass().getName()
     *     returns "[[[[[[[I"
     * </pre></blockquote>
```

Reply:
Sure. I thought about this and `name` should be the same for all types. the `canonicalName` implementation seems to use some reflection and other java processing to compute the value, while `name` is implemented natively so I figured it was more efficient to use that, but it is likely an over optimization if at all